fix usage of deprecated Lumberjack method

vindia-mb opened the pull request:
Fixes #1003

`Lumberjack#unit_of_work` was deprecated, see https://github.com/bdurand/lumberjack/commit/23c2e905759f0b52d8fb2d5f81b555e7180252da

I chose to replace the deprecated method with `tag(unit_of_work: SecureRandom.hex(6))` because this is as close to the original code as possible. However, given how this method is used in `Guard::Runner`, I think you could also just use `#context`, but I will leave that decision to the maintainers. I'm more than happy to change my PR to use `#context` instead.

vindia-mb committed:
fix usage of deprecated Lumberjack method

Lumberjack#unit_of_work was deprecated, see https://github.com/bdurand/lumberjack/commit/23c2e905759f0b52d8fb2d5f81b555e7180252da

fixes #1003

rymai commented:
> However, given how this method is used in Guard::Runner, I think you could also just use #context, but I will leave that decision to the maintainers. I'm more than happy to change my PR to use #context instead.

@vindia-mb Thanks the PR and sorry for the delay! I think using `#context` makes sense here. 👍

vindia-mb committed:
use LumberJack#context instead of #tag

vindia-mb commented:
@rymai Thanks for your feedback. I just updated my PR. I didn't see anything in your contribution guide line about using a single or multiple commits, so I just added the change to use `context` in an additional commit. Let me know if you prefer a single commit containing the final change.

rymai commented on lib/guard/runner.rb:
```diff
@@ -1,6 +1,7 @@
 # frozen_string_literal: true
 
 require "lumberjack"
+require "securerandom"
```
Do we still need this?

vindia-mb committed:
remove unneeded require of securerandom

this was needed for a previous implementation

scarver2 commented:
Thank you for fixing the Lumberjack deprecation.

vindia-mb commented:
Thanks for merging @rymai 🙏
